packages/babel-plugin-syntax-flow/src/index.ts

```diff
@@ -2,22 +2,30 @@ import { declare } from "@babel/helper-plugin-utils";
 
 export interface Options {
   all?: boolean;
-  enums?: boolean;
 }
 
 export default declare((api, options: Options) => {
   api.assertVersion(REQUIRED_VERSION(7));
 
   // When enabled and plugins includes flow, all files should be parsed as if
   // the @flow pragma was provided.
+  // @ts-expect-error Babel 7
   const { all, enums } = options;
 
   if (typeof all !== "boolean" && typeof all !== "undefined") {
     throw new Error(".all must be a boolean, or undefined");
   }
 
-  if (typeof enums !== "boolean" && typeof enums !== "undefined") {
-    throw new Error(".enums must be a boolean, or undefined");
+  if (process.env.BABEL_8_BREAKING) {
+    if (enums !== undefined) {
+      throw new Error(
+        "The .enums option has been removed and it's now always enabled. Please remove it from your config.",
+      );
+    }
+  } else {
+    if (typeof enums !== "boolean" && typeof enums !== "undefined") {
+      throw new Error(".enums must be a boolean, or undefined");
+    }
   }
```

Comment:
Now the `enums` option will have no effect, and we also explicitly throw an error here to align with ts's `allowDeclareFields`.